### ISSUE TITLE
[Assistive tech and device support] Alert and status messages aren't announced without receiving focus. (11.22.1)

### ISSUE THREAD

**Issue by shiragoodman** [department-of-veterans-affairs/va.gov-team#50941]
#### VFS team name
DGIB - My Education Benefits

#### VFS product name
Authenticated 22-1990e Transfer of Entitlement (TOE) Form Updates

#### Point of Contact/Reviewers
Brian DeConinck - @briandeconinck - Accessibility

*For more information on how to interpret this ticket, please refer to the [Anatomy of a Staging Review issue ticket](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/Anatomy-of-a-Staging-Review-Issue-ticket.2060320997.html) guidance on Platform Website.

---

### Platform Issue
Alert and status messages aren't announced without receiving focus.

### Issue Details
In Step 3 under "Choose how you want to get notifications," if there's no mobile number on file then a warning alert appears after the "Would you like to receive text message notifications" question and its radio buttons.

Placing the warning after the radio buttons, and not programmatically associating the warning to the question itself, means screen reader users are pretty likely to miss the warning entirely. Often, screen reader users will move through the questions in a form by pressing [Tab] to jump to the next question/interactive element. After selecting "Yes, send me text message notifications," the next [Tab] stop is "Finish this application later" -- skipping over the message that they've selected an option that doesn't work for them.

The actual impact of this is relatively low, since if they miss the warning they'll get an error message as soon as they try to advance to the next step. But it's an extra point of friction for screen reader users that would be good to avoid.

### VA.gov Experience Standard
[Category Number 11, Issue Number 22](https://depo-platform-documentation.scrollhelp.site/collaboration-cycle/VA.gov-experience-standards.1683980311.html)

### Other References
WCAG SC 4.1.3_AA

### Platform Recommendation
Does it make sense to change whether/how this question is displayed based on whether a mobile phone is on file? Since the warning and error message don't direct the user to any workflow for adding a mobile number, this feels like it could be an unnecessary dead end.

If that's not a viable option, then move the warning message inside the `fieldset` and associate the message with the `fieldset` element using `aria-labelledby`. That should prompt screen readers to announce the warning message at the same time as the `legend`. See this [CodePen from Steve Faulkner](https://codepen.io/stevef/pen/PQMaxr) for an example implementation.

---

### VFS Guidance

- Close the ticket when the issue has been resolved or validated by your Product Owner
- If your team has additional questions or needs Platform help validating the issue, please comment on the ticket
- Some feedback provided may be out of scope for your iteration of the product, however, Platform's OCTO leadership has stated that all identified issues need to be documented and it is still your responsibility to resolve the issue.
- If you do not believe that this Staging Review issue ticket is the responsibility of your team, comment below providing an explanation and who you believe is responsible. Please tag the Point of Contact/Reviewers. Governance team will research and will follow up.

**Comment by btruongafs** [comment]
Moved up warning message and added button to direct applicant to contact info page to input number. Issue resolved and deployed to staging.
